src/params/fields-param.js

```diff
@@ -10,13 +10,15 @@ class FieldsParam extends Str {
     if (typeof result === 'function') {
       result = result();
     }
-    return typeof result === 'string' ? objectFields.split(result) : result;
+    assert(Array.isArray(result));
```

Comment:
Because a function can return a `string`, Should we have this check in here? Or should we still support `strings` that are return from functions such as `fields: () => 'id,user(id,name)'`